### PR TITLE
Feat: Implemented EOA Constructor and get_evm_address

### PR DESCRIPTION
**Pull request by danilowhk**
Implemented the constructor for EOA (previously initialize()):

https://github.com/kkrt-labs/kakarot/blob/c0c80301d4dfe384247b423a3bc2726522b5d1a8/src/kakarot/accounts/eoa/externally_owned_account.cairo#L15

And get_evm_address function:
https://github.com/kkrt-labs/kakarot/blob/c0c80301d4dfe384247b423a3bc2726522b5d1a8/src/kakarot/accounts/eoa/externally_owned_account.cairo#L117

## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #216 #211 

## What is the new behavior?

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: crates/eoa/src/externally_owned_account.cairo [hunk 1]
```diff
@@ -1,3 +1,62 @@
...
+        let kakarot_token = IERC20Dispatcher { contract_address: native_token };
+        let infinite = BoundedInt::<u256>::max();
+        kakarot_token.approve(kakarot_address, infinite);
+        return;
+    }
+
+    #[external(v0)]
+    impl ExternallyOwnedAccount of super::IExternallyOwnedAccount<ContractState> {
+        fn get_evm_address(self: @ContractState) -> EthAddress {
+            return self.evm_address.read();
```
Comment (enitrat) [inline crates/eoa/src/externally_owned_account.cairo:48]:
```suggestion
            self.evm_address.read()
```

File: crates/eoa/src/externally_owned_account.cairo [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+        }
+
+        // @notice Empty bytecode needed for EXTCODE opcodes.
+        fn bytecode(self: @ContractState) -> Span<u8> {
+            return ArrayTrait::<u8>::new().span();
```
Comment (enitrat) [inline crates/eoa/src/externally_owned_account.cairo:53]:
```suggestion
            ArrayTrait::<u8>::new().span()
```

File: crates/eoa/src/externally_owned_account.cairo [hunk 3]
```diff
@@ -1,3 +1,62 @@
...
+        }
+
+        // @notice Empty bytecode needed for EXTCODE opcodes.
+        fn bytecode_len(self: @ContractState) -> u32 {
+            return 0;
```
Comment (enitrat) [inline crates/eoa/src/externally_owned_account.cairo:58]:
```suggestion
            0
```

File: crates/eoa/src/externally_owned_account.cairo [hunk 4]
```diff
@@ -1,3 +1,62 @@
...
+        ref self: ContractState,
+        native_token: ContractAddress,
+        kakarot_address: ContractAddress,
+        evm_address: EthAddress
+    ) {
+        self.evm_address.write(evm_address);
+        let kakarot_token = IERC20Dispatcher { contract_address: native_token };
+        let infinite = BoundedInt::<u256>::max();
+        kakarot_token.approve(kakarot_address, infinite);
+        return;
```
Comment (enitrat) [inline crates/eoa/src/externally_owned_account.cairo:42]:
```suggestion
```

**Comment by ClementWalter** [comment]
I do see that we had in the cairo0 impl a `get_evm_address` in each contract, but this sounds like a mistake because it's part of the share interface between CA and EOA. I don't see any reason not to put this in the account lib, which btw doesn't exist in this repo.

File: crates/eoa/src/externally_owned_account.cairo [hunk 5]
```diff
@@ -1,30 +1,60 @@
...
+        self.evm_address.write(evm_address);
+        let kakarot_token = IERC20Dispatcher { contract_address: native_token };
+        let infinite = BoundedInt::<u256>::max();
+        kakarot_token.approve(kakarot_address, infinite);
     }
 
     #[external(v0)]
     impl ExternallyOwnedAccount of super::IExternallyOwnedAccount<ContractState> {
-        /// Returns an empty span, required for the EXTCODE opcode
+        fn get_evm_address(self: @ContractState) -> EthAddress {
```
Comment (enitrat) [inline crates/eoa/src/externally_owned_account.cairo:46]: If we have a common library for both EOAs and Account contracts, I'd rather make an external component for that to share code and avoid maintaining twice the same thing. Should be ready next week-ish

**Comment by Eikix** [comment]
Putting this PR review on hold until we have Cairo Components in SSJ

**Comment by Eikix** [comment]
Closing this for now as it's outdated

**Pull request closed without merging**